### PR TITLE
Add Python packaging

### PR DESCRIPTION
**Pull request by svirpioj**
Changes:
* Add top-level Python setup scripts for easier installation of the wrapper
* Update CMakeLists
  * Add options for building only the executables or the wrapper
  * Use FindPython module (requires CMake >= 3.12)
  * Discard Python 2 support
* Add GitHub CI pipelines for tests and building and uploading Python wheels

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  build-and-test:
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            boost_platform: 20.04
+            boost_version: 1.73.0
+          - os: macos-11
```
Comment (vsiivola) [inline .github/workflows/tests.yaml:16]: Should we disable macos wheels and tests until we know it works? Could make a separate commit to disable those that is easily reverted?

**Comment by vsiivola** [comment]
👍 Nice work! Just thinking maybe better verify the macos stuff before enabling, or what do you think?

**Comment by vsiivola** [comment]
I guess the mac os packages are already downloadable from PYPI so could as well enable here. I'll merge.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (svirpioj) [reply .github/workflows/tests.yaml:16]: The tests on GitHub runners do work nicely - I tried yesterday also for macos-12 - but yes, there are problems with the wheels. Would be nice to know whether they work at least on intel Macs... I tried building the wheels also for macos with `arm64` architecture, but that fails with the following:

```
  File "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-8xgfyyc7/cp38-macosx_arm64/build/venv/lib/python3.8/site-packages/delocate/cmd/delocate_wheel.py", line 127, in main
374
    copied = delocate_wheel(
375
  File "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-8xgfyyc7/cp38-macosx_arm64/build/venv/lib/python3.8/site-packages/delocate/delocating.py", line 664, in delocate_wheel
376
    raise DelocationError(
377
delocate.delocating.DelocationError: Some missing architectures in wheel
378
Required arch arm64 missing from varikn/_varikn.so
379
Error: Command delocate-wheel --require-archs arm64 -w /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-8xgfyyc7/cp38-macosx_arm64/repaired_wheel -v /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-8xgfyyc7/cp38-macosx_arm64/built_wheel/varikn-0.1.dev1+g39d5ee5-cp38-cp38-macosx_11_0_arm64.whl failed with code 1. None
```

Couldn't find any hints for solving this.

Comment (vsiivola) [reply .github/workflows/tests.yaml:16]: Looks like nothing added arm64-compatible binaries to _varikn.so. I can try to look at this over the weekend.
